### PR TITLE
ADEN-2841 Extend prefooter_left to 970x250+728x90

### PR DESCRIPTION
**Pull request by drets**
Extend prefooter sizes:
- directGpt, remnantGpt: PR_LEFT: '300x250,728x90,970x250', PR_RIGHT: ''
- liftium: PR_LEFT: '300x250', PR_RIGHT: ''
- amazonMatch: PR_LEFT_BOXAD = ['3x2', '7x9', '9x2'];
- rubiconFastlane: PR_LEFT_BOXAD = [[300, 250], [728, 90], [970, 250]]; PR_RIGHT: ''

### CODE REVIEW

File: extensions/wikia/AdEngine/js/lookup/amazonMatch.js [hunk 1]
```diff
@@ -37,6 +38,10 @@ define('ext.wikia.adEngine.lookup.amazonMatch', [
 
 		slots = config[skin];
 
+		if (adContext.getContext().opts.overridePrefootersSizes) {
+			slots.PREFOOTER_LEFT_BOXAD = ['7x9', '9x2'];
```
Comment (jacekjursza) [inline extensions/wikia/AdEngine/js/lookup/amazonMatch.js]: What about 3x2 size?

Comment (drets) [inline extensions/wikia/AdEngine/js/lookup/amazonMatch.js]: :+1:

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+				if (slotMap.hasOwnProperty(slotName)) {
+					if (slotName.indexOf('TOP_LEADERBOARD') > -1) {
+						slotMap[slotName].size = '728x90';
+					}
+				}
 			}
 		}
+
+		if (context.opts.overridePrefootersSizes) {
+			slotMap.PREFOOTER_LEFT_BOXAD.size = '300x250,728x90,970x250';
```
Comment (jacekjursza) [inline extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js:23]: This is probably not for this story, but what do you guys think about keeping slot sizes in one place? I'm thinking about code readability. Sizes could be stored per slot, not per-provider. Current situation is that size-information is split into different places (often repeated) in codebase (different providers) and then more complicated by "overrideSizes" function. @nandy-andy @drets @fraszczakszymon

File: extensions/wikia/AdEngine/js/AdContext.js [hunk 3]
```diff
@@ -141,6 +141,19 @@ define('ext.wikia.adEngine.adContext', [
 			!context.targeting.wikiIsCorporate
 		);
 
+		// Override prefooters sizes
+		context.opts.overridePrefootersSizes =  !!(
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/AdContext.js:145]: What do you think about [adding a unit tests for those two cases as well](https://github.com/Wikia/app/blob/ADEN-2841-extend-prefooter-left/extensions/wikia/AdEngine/js/spec/AdContext.spec.js)?

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js (hunk 2 above)
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js:23]: I agree. While reading the code changes a red flag appeared in my head, connected to the fact how many dependencies to `ext.wikia.adEngine.adContext` we're introducing here. We should definitely rethink this approach. @drets, @jacekjursza who wants to create a ticket? :japanese_ogre:

Comment (jacekjursza) [inline extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js:23]: @nandy-andy I've created ticket: ADEN-2960

**Comment by nandy-andy** [comment]
LGTM :+1: As for the first step we'll refactor it within [ADEN-2960](https://wikia-inc.atlassian.net/browse/ADEN-2960).

Thanks for opening task, @jacekjursza, and adding few more unit tests, @drets! :smiley:

**Comment by nandy-andy** [comment]
The PHP unit tests aren't connected to the changes in this PR. They fail because of [broken Phalanx in dev environment](https://wikia-inc.atlassian.net/browse/SERVICES-1163).

**Comment by drets** [comment]
:whale:

**Pull request merged**